evennia/contrib/collab/commands.py

```diff
@@ -0,0 +1,817 @@
+                            "objects of type '%s'.{n" % self.create_type)
+            return
+        if not self.create_type in collab_settings.COLLAB_TYPES:
+            self.msg("{rType '%s' is not recognized.{n"
+                            % self.create_type)
+            return
+        self.player = getattr(self.caller, 'player', self.caller)
+        if not quota_check(self.player, self.create_type):
+            self.msg("{rYou have already met your quota for type '%s'.{n"
+                     % self.create_type)
+            return
+        thing = self.create_object()
+        if thing is None:
+            return
+        self.caller.db.last_created = thing
+        self.post_creation(thing)
+        self.success_message(thing)
+
+    def success_message(self, thing):
+        """
+        This is liable to change depending on what we're actually creating to
+        be more semantically useful.
+        """
+        self.msg("{gObject '%s' of type '%s' created with DBref #%s.{n" % (
+            thing.name, self.create_type, thing.id))
+
+    def post_creation(self, thing):
+        """
+        Perform any post-creation actions that should be done, like setting the
+        owner.
+        """
+        set_owner(self.caller, thing)
+
+    def perm_check(self, key, destination, location):
+        """
+        Verify that the user has permission to create an object with these
+        parameters. Raise an exception if not.
+        """
+        if destination:
+            if not collab_check(self.caller, destination, locks=['link_to']):
+                raise PermissionsError(
+                    "{rYou don't have permission to link to '%s'.{n"
+                    % destination)
+        if location:
+            if not collab_check(self.caller, location, locks=['create_in']):
+                raise PermissionsError(
+                    "{rYou can't create something in '%s'.{n" % location.name)
+
+    def create_object(self):
+        """
+        Perform the actual creation of the object, after some validation.
+        """
+        typeclass = collab_settings.COLLAB_TYPES[self.create_type]['typeclass']
+        try:
+            key, aliases = self.get_key_and_aliases()
+        except ValueError as err:
+            self.msg(str(err))
+            return
+        try:
+            destination = self.get_destination()
+        except ValueError as err:
+            self.msg(str(err))
+            return
+        try:
+            location = self.get_location()
+        except ValueError as err:
+            self.msg(str(err))
+            return
+        try:
+            self.perm_check(key, destination, location)
+        except PermissionsError as err:
+            self.msg(str(err))
+
+        return evennia.create_object(key=key, aliases=aliases, location=location,
+                                     destination=destination, typeclass=typeclass)
+
+
+class CmdCreate(BaseCreationCommand):
```

Comment:
If wanting to stick to the layout used by the default cmdsets, then the docstring should look like this: 

```python
Create an object. 

Usage:
  @create object name;object alias 1;alias2

Eventual further details here.
```